src/EventStore.Core/Services/Storage/StorageWriterService.cs

```diff
@@ -292,10 +306,11 @@ public class StorageWriterService<TStreamId> : IHandle<SystemMessage.SystemInit>
 
 						// when IsCommitted ExpectedVersion is always explicit
 						var expectedVersion = commitCheck.CurrentVersion + i;
+						var eventType = GetOrWriteEventType(evnt.EventType, ref logPosition);
```

Comment:
this line seems to be unnecessary since the event types are already in the `eventTypes` array

Reply:
:+1: i believe this was leftover from a previous iteration, its gone now